dulwich/refs.py

```diff
@@ -748,7 +795,10 @@ def read_loose_ref(self, name):
                 else:
                     # Read only the first 40 bytes
                     return header + f.read(40 - len(SYMREF))
-        except (FileNotFoundError, IsADirectoryError, NotADirectoryError):
+        except (OSError, UnicodeError):
```

Comment:
Since everything here is just dealing with bytes, what situations would you expect UnicodeError to be raised here?

Reply:
I just added it out of an abundance of caution — I'm not sure how it could happen in practice. In particular, I'm not sure how this behaves on Windows…